### PR TITLE
Remove BookRef and IssueRef (RIPD-1028)

### PR DESCRIPTION
**Pull request by miguelportilla**
Reviewers: @vinniefalco @scottschurr

### CODE REVIEW

File: src/ripple/protocol/Issue.h [hunk 1]
```diff
@@ -33,7 +33,7 @@ namespace ripple {
     When ByValue is `false`, this only stores references, and the caller
     is responsible for managing object lifetime.
 
-    @see Currency, AccountID, Issue, IssueRef, Book
+    @see Currency, AccountID, Issue, Issue, Book
```
Comment (scottschurr) [inline src/ripple/protocol/Issue.h]: Remove second occurrence of `Issue` in the comment.

**Comment by scottschurr** [comment]
The comment would be good to fix, but is not essential.

I'm assuming that the change is motivated by the realization that neither `Currency` nor `AccountID` are particularly heavy weight.  It makes a lot more sense for them to be carried by value than to have to worry about obscure lifetime-related crashes.  I agree.  And passing `Issue` and `Book` by `const&` is both smart and idiomatic.

Is the plan to remove the `ByValue` related template machinery in a later pull request?  That unused machinery really should be yanked, and I think removing it should be largely mechanical as well.

So, assuming that there is some plan in place for removing the now unused template machinery, I'm good with this code.  :+1:

**Comment by vinniefalco** [comment]
We definitely want to remove the template and clean out anything related to it.

**Comment by miguelportilla** [comment]
Thanks Scott, comment fixed.

**Comment by vinniefalco** [comment]
The template is still there?

**Comment by vinniefalco** [comment]
Note that this is not a trivial change

**Comment by miguelportilla** [comment]
@vinniefalco Which template?

**Comment by vinniefalco** [comment]
`IssueType<>`, `BookType<>`

**Comment by scottschurr** [comment]
I'm assuming that this pull request is waiting on the conversion of `BookType<true>` to a non-templated `Book` and , similarly, conversion of `IssueType<true>` to a non-templated `Issue`.  Personally, I'm okay with that happening in a separate pull request as long as there's a plan for that to happen.  Otherwise this code looks good to me.

**Comment by vinniefalco** [comment]
:+1:

**Pull request closed without merging**